### PR TITLE
Add support for '--syntax' flag in 'ocaml-mdx rule'

### PR DESCRIPTION
**Pull request by CraigFe**
`ocaml-mdx rule` accepts some options solely in order to pass them to the `ocaml-mdx test` rule that it generates (`--direction`, `--prelude` etc.), but it doesn't allow for passing the `--syntax` flag. This adds the `--syntax` flag.

This is very similar to #176, making me think this logic probably warrants a refactor to make it easier to understand; a pass-through option like `--test-options="--syntax=cram"` might work.

### CODE REVIEW

**Comment by CraigFe** [comment]
This is tested by #178. I can add a test directly here, but if the tests are just about to be refactored then it doesn't seem worthwhile to me.

**Comment by CraigFe** [comment]
Rebased over #171.

**Comment by CraigFe** [comment]
Rebased over #163.

**Comment by Julow** [comment]
I prefer having `rule` take the same options as `test`. So there is no different syntaxes between options that rule use and the ones it does not. In the future, `rule` may need to known the syntax.

I think we can merge this :)

What do you think @NathanReb, @samoht ?

**Comment by CraigFe** [comment]
> I prefer having `rule` take the same options as `test`. 

FWIW, I don't like this very much for two reasons:
 - The doc-strings for those options explain the use in `test` but not in `test`.
 - There are existing collisions that make this impossible: `--color`, `--quiet`, `--verbose` interact directly with the `rule` command and cannot be passed to `test` AFAICS. I think collisions like this are inevitable.

**Comment by CraigFe** [comment]
The test exists in #178; I was trying to avoid writing it here only to immediately delete it. Would you like it here too?

**Comment by NathanReb** [comment]
I really prefer when the feature/fix comes with its tests, especially since there's not much to discuss here but there might be in #178 (haven't reviewed it yet).

I totally trust that #178 is an improvement but given that it doesn't change `mdx`'s behaviour and that I'd like to release a new version soon, adding a test here would allow me to merge this with a free mind and be able to delay the review of #178 a bit (just a bit I promise)!

In short: if it's too much of a hassle don't bother, if it's easy enough that'd be great indeed!

**Comment by NathanReb** [comment]
Can you also add a changelog entry please?

**Comment by CraigFe** [comment]
Added test and changelog entry.
